### PR TITLE
[arm64] Addressing modes for SIMD

### LINKED ISSUE #67435: [arm64] Addressing mode for vectors
I noticed that we lose some perf in various `SpanHelpers` on arm64 due to missing addressing modes which brake pipelining, minimal repro:
```csharp
    Vector128<byte> Add(ref byte b1, ref byte b2, nuint offset) =>
        Vector128.LoadUnsafe(ref b1, offset) + 
        Vector128.LoadUnsafe(ref b2, offset);
```
Current codegen:
```asm
        add     x0, x1, x3
        ld1     {v16.16b}, [x0]
        add     x0, x2, x3
        ld1     {v17.16b}, [x0]
        add     v16.16b, v16.16b, v17.16b
        mov     v0.16b, v16.16b
```
Expected codegen:
```asm
        ldr     q16, [x1, x3]
        ldr     q17, [x2, x3]
        add     v16.16b, v16.16b, v17.16b
        mov     v0.16b, v16.16b
```
same for `[addr + imm]` e.g. `Vector128.LoadUnsafe(ref b2, 16)`

cc @tannergooding

### PR DESCRIPTION
**Pull request by EgorBo**
This PR implements addressing modes for SIMD vectors on arm64 (for now only loads with unscaled indices)
Closes https://github.com/dotnet/runtime/issues/67435

Benchmark:
```csharp
static byte[] _array1 = new byte[100000]; 
static byte[] _array2 = new byte[100000]; 
 
[Benchmark] 
public bool SequenceEqual() => _array1.AsSpan().SequenceEqual(_array2);
```

|        Method |             Toolchain |     Mean |     Error |    StdDev | Ratio |
|-------------- |---------------------- |---------:|----------:|----------:|------:|
| SequenceEqual |    /Core_Root/corerun | 3.635 us | 0.0031 us | 0.0027 us |  1.00 |
| SequenceEqual | /Core_Root_pr/corerun | **2.546 us** | 0.0224 us | 0.0210 us |  0.70 |

codegen diff example: https://www.diffchecker.com/wACQMKBT (`SpanHelpers.SequenceEqual`)

### CODE REVIEW

File: src/coreclr/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -2070,6 +2070,18 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
...
     }
+    else if (intrin.id == NI_AdvSimd_LoadVector128)
+    {
+        assert(intrin.numOperands == 1);
+
+        GenTree* addr = node->Op(1);
+        if (TryCreateAddrMode(addr, true, node) && IsSafeToContainMem(node, addr))
+        {
+            assert(addr->OperIs(GT_LEA));
+            assert(IsSafeToContainMem(node, addr));
```
Comment (jakobbotsch) [inline src/coreclr/jit/lowerarmarch.cpp:2081]: Leftover? This is checked in the if.

Comment (EgorBo) [reply src/coreclr/jit/lowerarmarch.cpp]: Yes, fixed, thanks!

**Comment by EgorBo** [comment]
@dotnet/jit-contrib PTAL, simple change, [diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1697438&view=ms.vss-build-web.run-extensions-tab) are small but cover SpanHelpers methods

**Review by jakobbotsch — APPROVED** [review]

**Comment by danmoseley** [comment]
Is this change sufficiently protected by an existing benchmark?

**Comment by EgorBo** [comment]
> Is this change sufficiently protected by an existing benchmark?

Yes, SpanHelpers are heavily used in many benchmarks and Regexes, we even have explicit benchmarks, e.g. https://github.com/dotnet/performance/blob/d7dac8a7ca12a28d099192f8a901cf8e30361384/src/benchmarks/micro/libraries/System.Memory/Span.cs#L48

**Pull request merged**